helper/certutil/helpers.go

```diff
@@ -156,50 +157,31 @@ func ParsePEMBundle(pemBundle string) (*ParsedCertBundle, error) {
 				parsedBundle.PrivateKeyBytes = pemBlock.Bytes
 			}
 		} else if certificates, err := x509.ParseCertificates(pemBlock.Bytes); err == nil {
-			switch len(certificates) {
-			case 0:
-				return nil, errutil.UserError{"pem block cannot be decoded to a private key or certificate"}
-
-			case 1:
-				if parsedBundle.Certificate != nil {
-					switch {
-					// We just found the issuing CA
-					case bytes.Equal(parsedBundle.Certificate.AuthorityKeyId, certificates[0].SubjectKeyId) && certificates[0].IsCA:
-						parsedBundle.IssuingCABytes = pemBlock.Bytes
-						parsedBundle.IssuingCA = certificates[0]
-
-					// Our saved certificate is actually the issuing CA
-					case bytes.Equal(parsedBundle.Certificate.SubjectKeyId, certificates[0].AuthorityKeyId) && parsedBundle.Certificate.IsCA:
-						parsedBundle.IssuingCA = parsedBundle.Certificate
-						parsedBundle.IssuingCABytes = parsedBundle.CertificateBytes
-						parsedBundle.CertificateBytes = pemBlock.Bytes
-						parsedBundle.Certificate = certificates[0]
-					}
-				} else {
-					switch {
-					// If this case isn't correct, the caller needs to assign
-					// the values to Certificate/CertificateBytes; assumptions
-					// made here will not be valid for all cases.
-					case certificates[0].IsCA:
-						parsedBundle.IssuingCABytes = pemBlock.Bytes
-						parsedBundle.IssuingCA = certificates[0]
-
-					default:
-						parsedBundle.CertificateBytes = pemBlock.Bytes
-						parsedBundle.Certificate = certificates[0]
-					}
-				}
-
-			default:
-				return nil, errutil.UserError{"too many certificates given; provide a maximum of two certificates in the bundle"}
-			}
+			certPath = append(certPath, &CertBlock{
+				Certificate: certificates[0],
+				Bytes:       pemBlock.Bytes,
+			})
 		}
+	}
 
-		if len(pemBytes) == 0 {
-			break
+	for i, certBlock := range certPath {
+		switch i {
+		case 0:
+			parsedBundle.Certificate = certBlock.Certificate
+			parsedBundle.CertificateBytes = certBlock.Bytes
+		case 1:
+			parsedBundle.IssuingCA = certBlock.Certificate
+			parsedBundle.IssuingCABytes = certBlock.Bytes
+		default:
+			parsedBundle.IssuingCAChain = append(parsedBundle.IssuingCAChain, certBlock.Certificate)
```

Comment:
I really don't know what the right answer is here.  My thought process was that you could always build the full chain if you wanted to but may want it separate.  The question may be, would you ever want the CA chain without the CA?  I would guess not.  Looking at how our public CA issues certs, they provide a file with the full chain.  

It would be a little more complicated to add the Issuing CA to the chain but not too bad.